extensions/chromium/pdfHandler.js

```diff
@@ -18,183 +18,243 @@ limitations under the License.
+        responseHeaders: [
+          {
+            header: "content-disposition",
+            values: ["attachment*"],
+          },
+        ],
+      },
+      action: ACTION_IGNORE_OTHER_RULES,
+    },
+    {
+      // Regular http(s) PDF requests.
+      condition: {
+        regexFilter: "^.*$",
+        // The viewer does not have the original request context and issues a
+        // GET request. The original response to POST requests is unavailable.
+        excludedRequestMethods: ["post"],
+        resourceTypes: ["main_frame", "sub_frame"],
+        responseHeaders: [
+          {
+            header: "content-type",
+            values: ["application/pdf", "application/pdf;*"],
+          },
+        ],
+      },
+      action: ACTION_REDIRECT_TO_VIEWER,
+    },
+    {
+      // Wrong MIME-type, but a PDF file according to the file name in the URL.
+      condition: {
+        regexFilter: "^.*\\.pdf\\b.*$",
+        // The viewer does not have the original request context and issues a
+        // GET request. The original response to POST requests is unavailable.
+        excludedRequestMethods: ["post"],
+        resourceTypes: ["main_frame", "sub_frame"],
+        responseHeaders: [
+          {
+            header: "content-type",
+            values: ["application/octet-stream", "application/octet-stream;*"],
+          },
+        ],
+      },
+      action: ACTION_REDIRECT_TO_VIEWER,
+    },
+    {
+      // Wrong MIME-type, but a PDF file according to Content-Disposition.
+      condition: {
+        regexFilter: "^.*$",
+        // The viewer does not have the original request context and issues a
+        // GET request. The original response to POST requests is unavailable.
+        excludedRequestMethods: ["post"],
+        resourceTypes: ["main_frame", "sub_frame"],
+        responseHeaders: [
+          {
+            header: "content-disposition",
+            values: ["*.pdf", '*.pdf"*', "*.pdf'*"],
+          },
+        ],
+        // We only want to match by content-disposition if Content-Type is set
+        // to application/octet-stream. The responseHeaders condition is a
+        // logical OR instead of AND, so to simulate the AND condition we use
+        // the double negation of excludedResponseHeaders + excludedValues.
+        // This matches any request whose content-type header is set and not
+        // "application/octet-stream". It will also match if "content-type" is
+        // not set, but we are okay with that since the browser would usually
+        // try to sniff the MIME type in that case.
+        excludedResponseHeaders: [
+          {
+            header: "content-type",
+            excludedValues: [
+              "application/octet-stream",
+              "application/octet-stream;*",
+            ],
+          },
+        ],
+      },
+      action: ACTION_REDIRECT_TO_VIEWER,
+    },
+  ];
+  for (const [i, rule] of addRules.entries()) {
+    rule.id = 15161 + i;
```

Comment:
Where does 15161 come from? Can we make it a constant to indicate its meaning?

Reply:
This is a "random" number, but fixed for predictability to make it easier to identify specific rules id ever needed. The value is the issue number linked this PR.

Every DNR ID needs to be unique, and be a positive number. So I would have to add at least 1 to the index. The exact value does not matter.

Reply:
I'm changing to `+1` + a comment. The implementation clears any pre-existing dynamic rules if any, so the exact IDs do not matter, as long as they are unique and at least 1.